linera-chain/src/data_types.rs

```diff
@@ -76,7 +76,7 @@ pub struct Block {
     pub previous_block_hash: Option<CryptoHash>,
 }
 
-impl Block {
+impl Proposal {
```

Comment:
Now a `BlockProposal` contains a `ProposalContent` which contains a `Proposal`.

Maybe call this type `ProposedBlock`?

Reply:
I didn't call it `ProposedBlock` on purpose b/c we have the `BlockProposal` which sounds basically equivalent that is the encompassing type. It seemed weird that `BlockProposal` contains `ProposedBlock`.

Reply:
But isn't a `BlockProposal` containing a `Proposal` even worse?

Maybe `BlockInput`? `BlockTemplate`? `BlockPlan`? :confused: